src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelPage/components/ExcelImportModal/index.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import * as styles from './styles.less';
+import { Spinner, OverlayPortal, Scrim, Button } from '@equinor/fusion-components';
+
+type ExcelImportModalProps = {
+    setSelectedFile: React.Dispatch<React.SetStateAction<File | null>>;
+    isProccessing: boolean;
+    isOpen: boolean;
+    setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const ExcelImportModal: React.FC<ExcelImportModalProps> = ({
+    setSelectedFile,
+    isProccessing,
+    isOpen,
+    setIsOpen,
+}) => {
+    const [selectedFileForUpload, setSelectedFileForUpload] = React.useState<File | null>(null);
+    const fileInput = React.useRef<HTMLInputElement>(null);
+    const [fileError, setFileError] = React.useState<string | null>(null);
+
+    React.useEffect(() => {
+        setSelectedFileForUpload(null);
+        setFileError(null);
+    }, [isOpen]);
+
+    const validateAndSetFile = React.useCallback((file: File) => {
+        const fileExtension = file.name.substr(file.name.lastIndexOf('.') + 1).toLowerCase();
+
+        if (fileExtension !== 'xlsx') {
+            setFileError('Fileformat not supported. Please again with a xlsx file');
+            setSelectedFileForUpload(null);
+            return;
+        }
+        setFileError(null);
+        setSelectedFileForUpload(file);
+    }, []);
+
+    const dragDropFileUpload = React.useCallback((e: React.DragEvent<HTMLDivElement>) => {
+        e.preventDefault();
+        if (!e.dataTransfer.items || e.dataTransfer.items[0].kind !== 'file') return;
+
+        const file = e.dataTransfer.items[0].getAsFile();
+        if (!file) return;
+        validateAndSetFile(file);
+    }, []);
+
+    const inputFileUpload = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        e.preventDefault();
+
+        const file = e.target.files ? e.target.files[0] : null;
+        if (!file) return;
+        validateAndSetFile(file);
+    }, []);
+
+    const fileInputClick = React.useCallback(() => {
+        fileInput?.current?.click();
+    }, [fileInput]);
+
+    const startProcessingSelectedFile = React.useCallback(() => {
+        setSelectedFile(selectedFileForUpload);
+    }, [selectedFileForUpload]);
+
+    const stopPropagationAndDefault = React.useCallback((e: React.DragEvent<HTMLDivElement>) => {
+        e.stopPropagation();
+        e.preventDefault();
+    }, []);
+
+    const closeModal = React.useCallback(() => {
+        !isProccessing && setIsOpen(false);
+    }, [isProccessing]);
+
+    return (
+        <OverlayPortal show={isOpen}>
```

Comment:
We don't use modals for stuff like this in Fusion. Modals/dialogs are only for alerts etc. Use a modal sidesheet for this